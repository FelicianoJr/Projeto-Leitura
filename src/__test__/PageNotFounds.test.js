import React from "react";
import { shallow } from "enzyme";
import PageNotFounds from "../components/PageNotFound";

const setup = () => {
  const component = shallow(<PageNotFounds />);

  return {
    component: component,
    h1: component.find("h1"),
    p: component.find("p")
  };
};

describe("<PageNotFounds/>", () => {
  it("should rende  h1", () => {
    const { h1 ,p} = setup();
    expect(h1.text()).toEqual("Erro 404");
    expect(p.text()).toEqual("Página não encontrada!");
    
  });

  it("should component Snapshot", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
