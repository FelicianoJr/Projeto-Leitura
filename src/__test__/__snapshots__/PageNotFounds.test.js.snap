// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFounds/> should component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFounds />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="display-4"
>
          Erro 404
</h1>,
        <p
          className="lead"
>
          Página não encontrada!
</p>,
        " ",
      ],
      "className": "jumbotron mt-2 mx-2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Erro 404",
          "className": "display-4",
        },
        "ref": null,
        "rendered": "Erro 404",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Página não encontrada!",
          "className": "lead",
        },
        "ref": null,
        "rendered": "Página não encontrada!",
        "type": "p",
      },
      " ",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="display-4"
>
            Erro 404
</h1>,
          <p
            className="lead"
>
            Página não encontrada!
</p>,
          " ",
        ],
        "className": "jumbotron mt-2 mx-2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Erro 404",
            "className": "display-4",
          },
          "ref": null,
          "rendered": "Erro 404",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Página não encontrada!",
            "className": "lead",
          },
          "ref": null,
          "rendered": "Página não encontrada!",
          "type": "p",
        },
        " ",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
